`test-utils` is not exported in `package.json`

**Describe the bug**

Jest 28 started supporting `exports` field in `package.json`. Imports from `react-intersection-observer/test-utils` are not working. Having:

```js
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';
```

Test is failing with:

```
Cannot find module 'react-intersection-observer/test-utils' from 'test.test.tsx'
```

**To Reproduce**

Put following in any test, which runs in Jest 28:

```js
import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';
```

Or run the following in Terminal in any project that has `react-intersection-observer` installed:

```bash
node -e 'require("react-intersection-observer/test-utils")'
```

Node.js fails to find an export:

```
node:internal/modules/cjs/loader:488
      throw e;
      ^

Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Package subpath './test-utils' is not defined by "exports" in /Users/aleks/bynder/paramount/node_modules/react-intersection-observer/package.json
    at new NodeError (node:internal/errors:371:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:440:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:692:3)
    at resolveExports (node:internal/modules/cjs/loader:482:36)
    at Function.Module._findPath (node:internal/modules/cjs/loader:522:31)
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:919:27)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at [eval]:1:1 {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

**Expected behavior**

Node.js could import `test-utils`.

**Additional context**

I tried react-intersection-observer 9.1.0 and 9.3.0.

Node.js v16.13.0

--- a/package.json
+++ b/package.json
@@ -7,7 +7,8 @@
   "module": "./dist/react-intersection-observer.mjs",
   "exports": {
     "require": "./dist/react-intersection-observer.js",
-    "default": "./dist/react-intersection-observer.modern.mjs"
+    "default": "./dist/react-intersection-observer.modern.mjs",
+    "test-utils": "./dist/test-utils.js"
   },
   "unpkg": "./dist/react-intersection-observer.umd.js",
   "typings": "./dist/index.d.ts",

